### ISSUE TITLE
[Bug]: Shard cost reduction from summoning mastery pool checkpoint applies permanently

### ISSUE THREAD

**Issue by mattbaron0** [MelvorIdle/melvoridle.github.io#2387]
### Describe the bug

The shard cost reduction from summoning mastery pool checkpoints (25% for T1 & T2 and 95% T3) remains applied even if mastery XP is spent below that checkpoint. So once you hit a checkpoint, shard costs for summoning tablets of the relevant tier are permanently reduced, regardless of whether you keep enough mastery XP to satisfy the checkpoints. I validated that the defect is not just visual -- actual shards used for crafting incorrectly include a reduced shard cost even when the checkpoint is not active.

Other mastery XP checkpoint bonuses for summoning appear to added and removed correctly when going above/below the checkpoint thresholds, only the shard cost reduction bonus appears to be treated permanently.

### Reproduction Steps

1. Acquire shard cost reduction from mastery pool checkpoints
2. Look at shard cost for relevant tablets.
3. Spend mastery pool XP below the checkpoint.
4. Look at the (unchanged) shard cost for relevant tablets.

### Expected behaviour

The shard cost reduction bonus should only apply when the relevant mastery pool checkpoints are enabled, or the Cape of Completion is worn.

### Save Link

https://melvoridle.com/save/wumFpoQP

### Which platforms are you experiencing this bug on?

- [ ] Chrome
- [X] Firefox
- [ ] Edge
- [ ] Safari
- [X] Mobile App (iOS)
- [ ] Mobile App (Android)
- [ ] Steam
- [ ] Other (Please Specify)

### Which version of the game are you experiencing this bug on?

v1.0.3

### Game Subversion

?1875

### Are you using any scripts or extensions?

No

**Comment by Flurryous** [comment]
Did you only test this with only the 95% pool or the 25% pool too? When I dropped below 25% my shard costs for t1 and t2 went up. However, it seems like dropping below 50% increased my t3 shard cost too, so it seems like the 50% mastery checkpoint is applying the shard reduction to t3 instead of the 95% pool. I was testing this on the steam version of the game.

**Comment by mattbaron0** [comment]
@Flurryous 

I definitely tested the 25% checkpoint and I saw the same behavior I reported above, though I might have made a mistake somehow.

Since reporting this I've completed summoning so I can't spend down my mastery any more to re-test or confirm your bug (though it looks like others have confirmed for you).

The bug you reported for T3 reductions applying incorrectly at the 50% checkpoint could easily be what I was seeing with T3 tablet costs, as I would have missed that with how I tested (I never rechecked T3 tablet prices after spending under the 50% checkpoint). I'm not as sure about the 25% checkpoint -- either something else is going on there, or I made some sort of mistake when I tested 25%.

I did originally write this up JUST for the 95% checkpoint issue, then tested 25% and rewrote it as an afterthought before submitting, so I wouldn't be surprised if I mixed something up when testing the 25% checkpoint.

**Comment by MelvorIdle** [comment]
I am unable to reproduce the non-visual aspect of this bug.

I can confirm in my testing with the Golbin Thief at Level 44 Mastery with 37% Mastery pool has a shard cost of 5 Red Shards. Performing 1 action removes 5 Red shards from my bank.

Spending my Mastery Pool down to 4%, I can see the visual cost of Red Shards is still at 5. However, performing 1 action will remove 6 Red shards from my bank.

I have tested with T2 and T3 Summons as well with the 95% checkpoint. The correct cost is still being removed regardless of what the visual element displays.

Is anyone able to help with providing a video of the issue by chance?

**Comment by mattbaron0** [comment]
@MelvorIdle 

I know the issue was more than visual. I have since completed summoning, so can't retest. 

I strongly suspect this might have actually been the issue that @Flurryous identified [#2409](https://github.com/MelvorIdle/melvoridle.github.io/issues/2409), though it looks like that was also closed as unable to reproduce. I saw the same code he provided a screenshot of which appears to be in error in `summoning.js`.
